### ISSUE TITLE
code issue

### ISSUE THREAD

**Issue by han508** [FreedomIntelligence/HuatuoGPT-II#3]
I found that WeightedRandomSampler has a update_dynamic_weight founction,but it was not used.Did you not use it in the training data?

**Comment by jymChen** [comment]
Hi @han508 ,
Our priority sampling algorithm does not require the use of the update_dynamic_weight function, as the data_priority is preset and remains unchanged. However, if there is a need to dynamically adjust the data_priority strategy during the training process, the update_dynamic_weight function can be utilized (which we have already implemented).

**Comment by han508** [comment]
Thank you for your reply. I have another problem. When converting the pre-trained corpus into instructions, I found that it would cause information loss to a large extent, for example, the original corpus was 3000 words, but the response after construction was only 300 words.

**Comment by han508** [comment]
To avoid this problem, can we just use question generated by ChatGPT and pre-training corpus to construct a <instruction, response>

**Comment by jymChen** [comment]
If the text is too lengthy, it is recommended to split the text into multiple paragraphs (corresponding to the segmentation operation in our data pipeline) and generate instructions for each paragraph separately.  Directly concatenating responses without rewriting is also feasible, although the effectiveness may vary.
